src/Components/Web.JS/src/Boot.WebAssembly.Common.ts

```diff
@@ -134,8 +147,20 @@ export async function startWebAssembly(components?: WebAssemblyComponentDescript
   api.invokeLibraryInitializers('afterStarted', [Blazor]);
 }
 
+export function hasWebAssemblyStarted(): boolean {
+  return hasStarted;
+}
+
+export function hasWebAssemblyStartedLoading(): boolean {
+  return platformLoadPromise !== undefined;
+}
+
+export function waitForBootConfigLoaded(): Promise<MonoConfig> {
+  return bootConfigPromise;
+}
+
 export function loadWebAssemblyPlatform(): Promise<void> {
-  platformLoadPromise ??= monoPlatform.load(options ?? {});
+  platformLoadPromise ??= monoPlatform.load(options ?? {}, resolveBootConfigPromise);
```

Comment:
Here you could have more complex method like
```js
async (config)=>{
    resolveBootConfigPromise(config)
    // delay wasm downloads by 1 second, until SSR is finished
    await new Promise(resolve => globalThis.setTimeout(resolve, 1000));
    // or 
    await untilSsrIsDone()
}
```

Reply:
That's a good suggestion. That change would really only impact stream rendering, since this code wouldn't run until after the initial SSR completes.

However, I think it's actually desirable to start loading WebAssembly resources while stream rendering is happening. Interactive components get activated when stream rendering completes, so loading WebAssembly assets in parallel with the stream response could significantly reduce the time to interactivity after the stream response ends. Also, from my testing, the stream rendering experience wasn't particularly negatively impacted by the background WebAssembly download. The main scenario that I found to be problematic on slow connections was:
* An auto mode component gets SSR'd, so WebAssembly bits start downloading
* The SSR completes, but WebAssembly resources haven't finished downloading yet
* The Blazor Server SignalR connection starts up, but its speed is severely impacted by the ongoing WebAssembly download

But even in that case, I don't think there's an obvious solution that addresses all customers' needs. Some customers might want to throttle the WebAssembly download for a better Blazor Server experience, but others might want to prioritize the WebAssembly download so that it's more likely to be cached on the next visit. I think these are considerations to take into account when we implement #48756, but it might be tricky to come up with a solution in this PR that works best for the majority of cases, especially since we don't yet have data on what customers want for their apps. Since #48756 was pushed to .NET 9, I would propose that we wait until then to implement APIs to configure this. Does that sound fine to you?

Reply:
Sure, I just wanted to hint at the possibility